content/6_Advanced/DP_More.mdx

```diff
@@ -17,10 +17,17 @@ frequency: 1
 		url="https://jeffreyxiao.me/blog/knuths-optimization"
 	 />
 	<Resource source="GCP" title="15.4.3 - Knuth's Optimization"/>
+	<Resource 
+		source="GFG" 
+		title="Knuth's Optimization in Dynamic Programming" 
+		url="https://www.geeksforgeeks.org/knuths-optimization-in-dynamic-programming/"
+	>
+		Good explanation + Proof of correctness
+	</Resource>
 </Resources>
 
 Knuth's optimization is, in a way, a special case of Range DP. In general, it is used to solve DP problems with the following transition - $\texttt{dp}[i][j]$ = $\texttt{cost}[i][j] + min_{i≤k<j}(\texttt{dp}[i][k] + \texttt{dp}[k+1][j])$.  
-Further, the cost function must be monotonic on lattice intervals and it must satisfy the quadrangle inequality, i.e., for all $a\leq b\leq c \leq d$:  
+Further, the cost function must be monotonic and it must satisfy the quadrangle inequality, i.e., for all $a\leq b\leq c \leq d$:  
```

Comment:
https://math.stackexchange.com/questions/321812/generalizing-monotonicity-to-2d
since cost is a 2D array, how does monotonicity work?

Reply:
Understood where you're coming from. ig it isn't that clear in the article, so I'll amend it. 

Basically, when I first saw this term used to describe the condition cost(a, d) >= cost(b, c), my interpretation was that in most scenarios the cost function is sort of a difference of 2 functions, like something of the form cost(x, y) = g(f(y) - f(x)), where g and f are monotonically increasing. Because in every example problem I saw, and even in the original problem for which this optimization was first developed, this was normally the case. Thinking of the cost as a function rather than a 2D array may help in intuitively understanding it. But I agree it isn't clear, so I've removed that term from the article to avoid confusion.